qemu/qemu_unix.go

```diff
@@ -510,9 +511,26 @@ func (q *qemu) setConfig(rconfig *types.RunConfig) error {
 	q.addOption("-device", "virtio-rng-pci")
 
 	disks := 1
-	for _, hostDir := range rconfig.VirtfsShares {
-		q.addOption("-virtfs", fmt.Sprintf("local,path=%s,mount_tag=%d,security_model=none,multidevs=remap,id=hd%d", hostDir, disks, disks))
-		disks++
+
+	if len(rconfig.VirtfsShares) > 0 {
+
+		// if using virtual fs shares sort first.
```

Comment:
We don't actaully need to sort the virtfs shares. As long as there is a fixed correspondence between virtual IDs and host dirs (which there is, due to the sorting done in `AddVirtfsShares`), we can simply iterate over rconfig.VirtfsShares, as in:
```
for diskId, hostDir := range rconfig.VirtfsShares {
		q.addOption("-virtfs", fmt.Sprintf("local,path=%s,mount_tag=%s,security_model=none,multidevs=remap,id=hd%s", hostDir, diskId, diskId))
}
```
The order in which these `-virtfs` command line options are added to the qemu command lone may vary, but that doesn't matter.